src/dreammaker/lexer.rs

```diff
@@ -630,9 +634,14 @@ impl<'ctx> HasLocation for Lexer<'ctx> {
 impl<'ctx> Lexer<'ctx> {
     /// Create a new lexer from a byte stream.
     pub fn new<I: Into<Cow<'ctx, [u8]>>>(context: &'ctx Context, file_number: FileId, input: I) -> Self {
+        let mut cow = input.into();
+        if has_bom(&cow) {
+            cow = Cow::from(cow[3..].to_owned());
```

Comment:
Should be possible to subslice a Borrowed or to erase the first three bytes of an Owned, rather than always reallocating

Reply:
What do you mean with erase? Set the BOM to null bytes? That would mess with unless I make the rest of the lexer ignore null bytes.

Reply:
definitely a micro-optimization but
```rust
Cow::Owned(mut o) => { o.drain(..3); Cow::Owned(o) }
```

Reply:
Ah I didn't realize that the `Owned` of a `Cow<[T]>` is a `Vec`. That makes sense yeah.